app/upgrades/v4/upgrades.go

```diff
@@ -0,0 +1,52 @@
+package v4
+
+import (
+	"context"
+	"strings"
+
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+
+	"github.com/milkyway-labs/milkyway/v3/app/keepers"
+)
+
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configuration module.Configurator,
+	keepers *keepers.AppKeepers,
+) upgradetypes.UpgradeHandler {
+	return func(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		vm, err := mm.RunMigrations(ctx, configuration, fromVM)
+		if err != nil {
+			return nil, err
+		}
+
+		// Unwrap the context
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+		// Remove all the markets except from the TIA/USD market
+		markets, err := keepers.MarketMapKeeper.GetAllMarkets(sdkCtx)
+		if err != nil {
+			return nil, err
+		}
+
+		for ticker, market := range markets {
```

Comment:
Let's sort map keys first and iterate the map with keys.

Reply:
Done!